### PR TITLE
Remove member group checks and check for permission instead

### PR DESCRIPTION
**Pull request by Rieven**
### Changes
Remove group checks in especially onboarding and general checks for groups at certain place in the code base.

### Issue link
https://github.com/minvws/nl-kat-coordination/issues/1050

Closes https://github.com/minvws/nl-kat-coordination/issues/1050

### Proof
_Please add some proof of your working change here, unless this is not required (e.g. this PR is trivial)._

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/onboarding/view_helpers.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 from django.utils.translation import gettext_lazy as _
 from tools.view_helpers import StepsMixin
 
+ONBOARIDNG_PERMS = ["tools.can_scan_organization", "tools.can_set_clearance_level", "tools.can_enable_disable_boefje"]
```
Comment (ammar92) [inline rocky/onboarding/view_helpers.py:6]:
```suggestion
ONBOARIDNG_PERMISSIONS = ["tools.can_scan_organization", "tools.can_set_clearance_level", "tools.can_enable_disable_boefje"]
```

Comment (ammar92) [reply rocky/onboarding/view_helpers.py:6]: You're also better of with a tuple instead of a list

**Review by ammar92 — COMMENTED** [review]
A few minor remarks, looks good otherwise

File: rocky/onboarding/view_helpers.py (hunk 1 above)
Comment (Rieven) [reply rocky/onboarding/view_helpers.py]: I see there is a typo also in ONBOARDING...fix it in [c82ccb9](https://github.com/minvws/nl-kat-coordination/pull/1275/commits/c82ccb92ae3e36e8f2dcd53edcc24a89d98e7804)

**Review by ammar92 — DISMISSED** [review]

**Comment by TwistMeister** [comment]
I noticed that when going through the onboarding as a superuser the "create report" steps are skipped. Is this intentional? I can understand that it could be, but for me personally its a bit undesirable because I liked the option to create the report in the onboarding flow to populate KAT with some basic objects and tasks. And of course you could skip this if you wanted. So I feel it's less flexible now than it was before, and am wondering if that's really what we want.

**Comment by TwistMeister** [comment]
@Rieven I verified and it was intentional for superusers to follow the whole onboarding flow, so it's regression when a part of it gets skipped.

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* Original onboarding flow has been restored :+1:

File: rocky/tools/management/commands/setup_dev_account.py [hunk 2]
```diff
@@ -45,6 +45,7 @@ def handle(self, *args, **options):
                 "view_organizationmember",
                 "add_organizationmember",
                 "change_organization",
+                "can_scan_organization",
```
Comment (dekkers) [inline rocky/tools/management/commands/setup_dev_account.py:48]: We need to create a migration to also add this permission to the admin group in existing installations.

File: rocky/account/forms/account_setup.py [hunk 3]
```diff
@@ -159,7 +159,7 @@ def save(self, **kwargs):
                 status=OrganizationMember.STATUSES.ACTIVE,
             )
             member.groups.add(selected_group.id)
-            if member.is_admin or self.user.is_superuser:
+            if member.has_perm("change_organizationmember") or self.user.is_superuser:
```
Comment (dekkers) [inline rocky/account/forms/account_setup.py:162]: Why does a new member that can change organization members automatically get scan level 4? Wouldn't it be better to have the user set this explicitly? And if we want to have this way, shouldn't we this and probably make this also clear in the user interface?

This is something we can change in a later PR.

Comment (Rieven) [reply rocky/account/forms/account_setup.py:162]: Here admins have by default trusted and acknowledged clearance levels by L4 so they can start scans but do not have clearance permission on objects, so redteamers are the only ones that can increase clearance levels on objects. 

We can refactor this part to put defaults at member creation when member is an admin.

**Review by dekkers — APPROVED** [review]

**Pull request merged**